FilteringApp/FilteringApp.ConsoleUI/Program.cs

```diff
@@ -1,58 +1,114 @@
+        {
+            int[] array = CycleArrayReading();
+            DisplayMinMax(array.FindMinAndMax());
+            Console.ReadLine();
+        }
+
+        static UserInput ValidateUserInput()
+        {
+            int[] array = CycleArrayReading();
+            var intValue = CycleFilteringValueReading();
+            var userInput = new UserInput(array, intValue);
+
+            return userInput;
+        }
+
+        private static void InstanceCountItself()
+        {
+            CreateInstances();
+            Console.WriteLine($"Number of living instances of SelfInstanceCount Class is:  { SelfInstanceCount.activeCount}");
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+            GC.Collect();
+            Console.ReadLine();
+        }
+
+        private static void CreateInstances()
+        {
+            new SelfInstanceCount();
+            new SelfInstanceCount();
+        }
+
+        public static int[] CycleArrayReading()
+        {
+            try
             {
-                Console.WriteLine("Please provide INT numbers for the array, separated by commas (for example like 1,2,3,4,...,12321,21433).");
-                var stringArray = Console.ReadLine().Split(',');
-                arrayPackage = ArrayUtils.CheckArray(stringArray);
-                if (arrayPackage.IsValid == false)
-                {
-                    Console.WriteLine(arrayPackage.Error);
-                }
-                else
-                {
-                    isArrayValid = arrayPackage.IsValid;
-                }
+                return ReadUserArray();
             }
-            var intValue = 0;
-            var isFilteringValueValid = false;
-            var userValueParseResults = new UserValueParseResult(intValue);
-            while (isFilteringValueValid == false)
+            catch (Exception ex)
             {
-                Console.WriteLine("Please provide an INT value for filtering the array.");
-                var value = Console.ReadLine();
-                var intValueObject = ArrayUtils.CheckFilteringValue(value);
-
-                if (intValueObject.IsValid == false)
-                {
-                    Console.WriteLine(userValueParseResults.Error);
-                }
-                else
-                {
-                    intValue = intValueObject.Value;
-                    isFilteringValueValid = true;
-                }
+                Console.WriteLine(ex.Message);
+
+                return CycleArrayReading();
             }
-            var userInput = new UserInput(arrayPackage.Array, intValue);
-            var filteredValue = ArrayUtils.FilterOutResults(userInput);
-            DispplayFilteringResults(filteredValue);
-            DisplaySummingResults(ArrayUtils.AddArrayElements(arrayPackage.Array));
         }
 
-        public static void DispplayFilteringResults(List<int> result)
+        public static int CycleFilteringValueReading()
```

Comment:
Why public?

Reply:
Same for next methods

Reply:
good question,  all of them made private now.